bulbs/ads/views.py

```diff
@@ -0,0 +1,61 @@
+import json
+import sys
+
+from django.core.urlresolvers import resolve, Resolver404
+from django.http import HttpResponse, HttpResponseForbidden, HttpResponseNotFound
+from django.views.decorators.csrf import csrf_exempt
+from django.test.client import RequestFactory
+
+from .models import TargetingOverride
+
+
+if sys.version_info.major == 3:
+    import urllib.parse as urlparse
+else:
+    from urlparse import urlparse
+
+
+@csrf_exempt
+def targeting(request):
+    if not (request.user and request.user.has_perm("ads.change_targetingoverride")):
+        return HttpResponseForbidden(
+            '{"detail": "You do not have permission to perform this action."}'
+        )
+
+    url = request.GET.get("url")
+    if url is None:
+        return HttpResponseNotFound()
+
+    if url.startswith("http://") or url.startswith("https://"):
+        parsed = urlparse(url)
+        url = parsed.path
+
+    try:
+        view, args, kwargs = resolve(url)
+    except Resolver404:
+        return HttpResponseNotFound()
+
+    req = RequestFactory().get(url)
+    kwargs['request'] = req
+    response = view(*args, **kwargs)
+    if response.status_code != 200:
+        return HttpResponseNotFound()
+
+    targeting = response.context_data.get('targeting', {})
+
+    if request.method == "POST":
```

Comment:
What is this used for?  Curious when you'd ever make a `POST` request where this comes into play